.github/scripts/trymerge.py

```diff
@@ -1060,34 +1074,39 @@ def find_matching_merge_rule(pr: GitHubPR,
         if len(approvers_intersection) == 0 and len(rule_approvers_set) > 0:
             if reject_reason_score < 10000:
                 reject_reason_score = 10000
-                reject_reason = (f"Matched rule {rule_name}, but PR #{pr.pr_num} was not reviewed yet by any of: " +
-                                 f"{', '.join(list(rule_approvers_set)[:5])}{', ...' if len(rule_approvers_set) > 5 else ''}")
+                reject_reason = "\n".join((
+                    f"Approval needed from one of the following (Rule '{rule_name}'):",
+                    f"{', '.join(list(rule_approvers_set)[:5])}{', ...' if len(rule_approvers_set) > 5 else ''}"
+                ))
             continue
+
+        # Does the PR pass the checks required by this rule?
         mandatory_checks = rule.mandatory_checks_name if rule.mandatory_checks_name is not None else []
         checks = get_combined_checks_from_pr_and_land_validation(pr, land_check_commit)
         required_checks = filter(lambda x: force is False or "CLA Check" in x, mandatory_checks)
         [pending_checks, failed_checks] = categorize_checks(checks, required_checks)
 
+        hud_link = f"https://hud.pytorch.org/{pr.org}/{pr.project}/commit/{pr.last_commit()['oid']}"
```

Comment:
Why this and not
```suggestion
        hud_link = f"https://hud.pytorch.org/pr/{pr.number}"
```

Reply:
To make it work for other projects I guess (but we can use a shortcut of `f"{pr.org}/{pr.project}"=="pytorch/pytorch"`), and also to make it work for landchecks as well

Reply:
Other than that being the original url, I like the commit specific links more for debugging purposes.  This message is calling out that there _was_ an error, and the commit specific version will always point to that error.

If the person pushes a new change that passes checks, the failing commit link will no longer make sense. We already offer other links to show the status of the latest changes so I think this would be preferable